apps/challenges/aws_utils.py

```diff
@@ -1487,3 +1584,287 @@ def challenge_approval_callback(sender, instance, field_name, **kwargs):
+            
+            return response
+        except ClientError as e:
+            logger.exception(e)
+            return
+        
+
+
+@app.task    
+def delete_eks_vpc(challenge):
+    """
+    Remove VPC and associated Subnets, InternetGateways, RouteTables and SecurityGroups from AWS
+    calls delete_eks_nodegroup function
+
+    Arguments:
+        instance {<class 'django.db.models.query.QuerySet'>} -- instance of the model
+    """
+    from .models import ChallengeEvaluationCluster
+    from .serializers import ChallengeEvaluationClusterSerializer
+    from .utils import get_aws_credentials_for_challenge
+
+    for obj in serializers.deserialize("json", challenge):
+        challenge_obj = obj.object
+    challenge_evaluation_cluster = (
+        ChallengeEvaluationCluster.objects.get(challenge=challenge_obj)
+    )
+    vpcid = challenge_evaluation_cluster.vpc_id
+    if not vpcid:
+        return
+    challenge_aws_keys = get_aws_credentials_for_challenge(
+        challenge.pk
+    )
+    ec2 = get_boto3_resource("ec2", challenge_aws_keys)
+    client = ec2.meta.client
+    
+    vpc = ec2.Vpc(vpcid)
+    # detach default dhcp_options if associated with the vpc
+    dhcp_options_default = ec2.DhcpOptions('default')
+    if dhcp_options_default:
+        dhcp_options_default.associate_with_vpc(
+            VpcId=vpc.id
+        )
+    # detach and delete all gateways associated with the vpc
+    for gw in vpc.internet_gateways.all():
+        try:
+            vpc.detach_internet_gateway(InternetGatewayId=gw.id)
+            gw.delete()
+        except ClientError as e:
+            logger.exception(e)
+            return
+    # delete all route table associations
+    for rt in vpc.route_tables.all():
+        for rta in rt.associations:
+            if not rta.main:
+                try:
+                    rta.delete()
+                except ClientError as e:
+                    logger.exception(e)
+                    return
+    # delete any instances
+    for subnet in vpc.subnets.all():
+        for instance in subnet.instances.all():
+            try:
+                instance.terminate()
+            except ClientError as e:
+                logger.exception(e)
+                return
+    # delete our endpoints
+    for ep in client.describe_vpc_endpoints(
+            Filters=[{
+                'Name': 'vpc-id',
+                'Values': [vpcid]
+            }])['VpcEndpoints']:
+        try:
+            client.delete_vpc_endpoints(VpcEndpointIds=[ep['VpcEndpointId']])
+        except ClientError as e:
+            logger.exception(e)
+            return
+    # delete our security groups
+    for sg in vpc.security_groups.all():
+        try:
+            if sg.group_name != 'default':
+                sg.delete()
+        except ClientError as e:
+            logger.exception(e)
+            return
+    # delete any vpc peering connections
+    for vpcpeer in client.describe_vpc_peering_connections(
```

Comment:
`vpcpeer` to `vpc_peer` we might not need this if you use `delete_vpc`

Reply:
Everything associated to the VPC must be deleted/deallocated before we can delete the VPC using `delete_vpc`